src/h5cli/cli.py

```diff
@@ -107,6 +109,40 @@ def do_clear(self, args):
     def do_exit(self, args):
         return True
 
+    @options([
+        make_option('-s', '--shape', action='store_true', help='print the shape of datasets')
+    ])
+    def do_tree(self, args, opts=None):
+        """list contents of groups in a tree-like format."""
+        global __groupcount
+        global __datasetcount
```

Comment:
Why are these globals?

Reply:
In order to reference them from both the inner functions and the print statement, I found that at least the easiest way of doing so was to mark them as global. I am open to other suggestions, but wanted to avoid doing multiple tree traversals.

I'm using a library for printing the tree, and it's not as well documented as I would like, but it seems to work. It takes callables and does not handle any alternative values returned.

Reply:
ping @lphk92